pyriemann/spatialfilters.py

```diff
@@ -562,24 +576,40 @@ def fit(self, X, y=None):
         self : AJDC instance
             The AJDC instance.
         """
-        # estimation of cospectra
+        # definition of params for Welch's method
         cospcov = est.CospCovariances(
             window=self.window,
             overlap=self.overlap,
             fmin=self.fmin,
             fmax=self.fmax,
             fs=self.fs)
-        cosp = cospcov.transform(X)
-        self.n_channels_ = cosp.shape[1]
-        self.freqs_ = cospcov.freqs_
+        # estimation of cospectra on subjects and conditions
+        cosp = []
+        for s in range(len(X)):
+            cosp_ = cospcov.transform(X[s])
+            if s == 0:
+                n_conditions = cosp_.shape[0]
+                self.n_channels_ = cosp_.shape[1]
+                self.freqs_ = cospcov.freqs_
+            else:
+                if n_conditions != cosp_.shape[0]:
+                    raise ValueError('Unequal number of conditions')
+                if self.n_channels_ != cosp_.shape[1]:
+                    raise ValueError('Unequal number of channels')
+            cosp.append(cosp_)
+        cosp = numpy.transpose(numpy.array(cosp), axes=(0, 1, 4, 2, 3))
+
+        # trace-normalization of cospectra, Eq(3) in [2]
+        cosp = self._normalize_trace(cosp)
+        # average of cospectra across subjects, Eq(7) in [2]
+        cosp = numpy.mean(cosp, axis=0, keepdims=False)
```

Comment:
Is there a study showing the improvement of gBSS when using a Riemannian mean across subjects? @Marco-Congedo

Reply:
I am not aware of such a study. Cheers